packages/ui/src/components/Form/dataFormat/DataFormatEditor.tsx

```diff
@@ -78,6 +80,8 @@ export const DataFormatEditor: FunctionComponent<DataFormatEditorProps> = (props
     [entitiesContext, dataFormatCatalogMap, props.selectedNode.data?.vizNode],
   );
 
+  if (props.formMode === FormTabsModes.USER_MODIFIED && !isDefined(selectedDataFormatOption)) return null;
```

Comment:
This is an interesting way to check whether this field should be shown or not, although I expected to control it through the parent, I think we can move forward for now with this approach. My recommendation would be to have a unit test explaining this situation.

Reply:
Thanks for the review!
Can you share you thoughts on how to control it through the parent?

Reply:
Since the `CanvasFormTabs` component knows that a given step needs to render a `DataFormatEditor`, perhaps we could use the `DataFormatService` to check whether a language was provided or not